packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.spec.tsx

```diff
@@ -104,68 +105,71 @@ describe('<ArrayInput />', () => {
     it('should clone each input once per value in the array', () => {
         render(
             <AdminContext dataProvider={testDataProvider()}>
-                <SimpleForm
-                    onSubmit={jest.fn}
-                    defaultValues={{
-                        arr: [
-                            { id: 123, foo: 'bar' },
-                            { id: 456, foo: 'baz' },
-                        ],
-                    }}
-                >
-                    <ArrayInput resource="bar" source="arr">
-                        <SimpleFormIterator>
-                            <NumberInput source="id" />
-                            <TextInput source="foo" />
-                        </SimpleFormIterator>
-                    </ArrayInput>
-                </SimpleForm>
+                <ResourceContextProvider value="bar">
+                    <SimpleForm
+                        onSubmit={jest.fn}
+                        defaultValues={{
+                            arr: [
+                                { id: 123, foo: 'bar' },
+                                { id: 456, foo: 'baz' },
+                            ],
+                        }}
+                    >
+                        <ArrayInput source="arr">
+                            <SimpleFormIterator>
+                                <NumberInput source="id" />
+                                <TextInput source="foo" />
+                            </SimpleFormIterator>
+                        </ArrayInput>
+                    </SimpleForm>
+                </ResourceContextProvider>
             </AdminContext>
         );
         expect(
-            screen.queryAllByLabelText('resources.bar.fields.arr.id')
+            screen.queryAllByLabelText('resources.bar.fields.id')
```

Comment:
This doesn't feel right and it's a BC. I know that I asked you to revert the change that causes this, but it was my mistake. 

translation messages of inputs in arrayinputs should have the arrayinput source in it - just not the index. 

So with this setup:

```
<ArrayInput source="arr">
   <SimpleFormIterator>
      <NumberInput source="id" />
   </SimpleFormIterator>
</ArrayInput>
```

The default label of all the `id` number inputs should be `resources.bar.fields.arr.id` (but not `resources.bar.fields.arr.0.id` or `resources.bar.fields.id`)